src/impl_views/conversions.rs

```diff
@@ -56,6 +57,36 @@ where
     }
 }
 
+
+/// Methods specific to `ArrayView0`.
+///
+/// ***See also all methods for [`ArrayView`] and [`ArrayBase`]***
+///
+/// [`ArrayBase`]: struct.ArrayBase.html
+/// [`ArrayView`]: struct.ArrayView.html
+impl<'a, A> ArrayView<'a, A, Ix0> {
+    /// Consume the view and returns a reference to the single element in the array.
+    ///
+    /// The lifetime of the returned reference matches the lifetime of the data
+    /// the array view was pointing to.
+    ///
+    /// ```
+    /// use ndarray::{arr0, Array0};
+    ///
+    /// // `Foo` doesn't implement `Clone`.
+    /// #[derive(Debug, Eq, PartialEq)]
+    /// struct Foo;
+    ///
+    /// let array: Array0<Foo> = arr0(Foo);
+    /// let view = array.view();
+    /// let scalar: &Foo = view.into_scalar();
+    /// assert_eq!(scalar, &Foo);
+    /// ```
+    pub fn into_scalar(self) -> &'a A {
```

Comment:
the ArrayView doesn't need to be consumed to allow this (infuriating difference between shared and mut for IndexLonger). I guess one could just as well make this a &self method, it makes difference for non-copy ArrayViews?

Reply:
I thought about it, but then I opted to keep the signature consistent with the other two (`Array`/`ArrayViewMut`). Given that it's named `into_*`, it sounded preferable to consume the input.
On the other hand, taking `&self` makes it more convenient for `ArrayView` because you can keep using `self` afterwards.

I don't know, I would lean on the side of symmetry in this case.

Reply:
ok, that works.